readers/mongodb/messages_test.go

```diff
@@ -117,13 +136,13 @@ func TestReadSenml(t *testing.T) {
 		},
 		"read message last page": {
 			chanID: chanID,
-			offset: 40,
-			limit:  10,
+			offset: 95,
+			limit:  limit,
```

Comment:
Maybe we can remove this `limit` field from this struct type and use `tc.page.Limit` in line 274. It's an unecessary repetition.

Reply:
`tc.page.limit` represents the response while `limit` represents a parameter of the request. We can't use it as the same value but what we can maybe add is a test like it's done for `Total`. Something like: `assert.Equal(t, tc.page.Limit, result.Limit, fmt.Sprintf("%s: expected %v got %v", desc, tc.page.Limit, result.Limit))`

Reply:
I understand. AFAIK, the value of `tc.limt` and `tc.Page.Limit` is always the same. If they are guaranteed to be same, one should be removed.

Besides, maybe I'm missing something, but the values of `tc.Page.Limit` and `tc.Page.Offset` are not used in tests. We should either remove them or use them.

Reply:
Yes, you are right, it's not used. It's also better if we pass all filters in the `page`